core/src/main/scala/org/apache/spark/status/api/v1/ApplicationListResource.scala

```diff
@@ -69,6 +69,8 @@ private[spark] object ApplicationsListResource {
       attempts = app.attempts.map { internalAttemptInfo =>
         new ApplicationAttemptInfo(
           attemptId = internalAttemptInfo.attemptId,
+          startTimeEpoch = internalAttemptInfo.startTime,
```

Comment:
@ajbozarth thank you for your patience, this is pretty close now. I apologize, I haven't really been paying attention to the rest of this change. It is conceptually fine but actually, this part of the change is not necessary. You've already got `Date` objects encoding the the time in this object, which is really a fancy wrapper around ms since the epoch. `getTime` gets you the same value you're adding here already. So I don't think it's necessary to actually store all these values a second time. 

Of course you still need to expose these values as an integer `Long` as properties of the class, so that JSON will contain the second ms representation of them. I think that only requires adding getter methods with the right name to the class objects below, not actually adding redundant fields. How about that?